tests/floris_interface_test.py

```diff
@@ -23,15 +23,90 @@ def test_calculate_wake():
     that the yaw angles are correctly set in subsequent calls to calculate_wake.
     """
     fi = FlorisInterface(configuration=YAML_INPUT)
-    yaw_angles = 20 * np.ones((fi.floris.flow_field.n_wind_directions, fi.floris.flow_field.n_wind_speeds, fi.floris.farm.n_turbines))
+    yaw_angles = 20 * np.ones(
+        (
+            fi.floris.flow_field.n_wind_directions,
+            fi.floris.flow_field.n_wind_speeds,
+            fi.floris.farm.n_turbines
+        )
+    )
     fi.calculate_wake(yaw_angles=yaw_angles)
     assert fi.floris.farm.yaw_angles == yaw_angles
 
-    yaw_angles = np.zeros((fi.floris.flow_field.n_wind_directions, fi.floris.flow_field.n_wind_speeds, fi.floris.farm.n_turbines))
+    yaw_angles = np.zeros(
+        (
+            fi.floris.flow_field.n_wind_directions,
+            fi.floris.flow_field.n_wind_speeds,
+            fi.floris.farm.n_turbines
+        )
+    )
     fi.calculate_wake(yaw_angles=yaw_angles)
     assert fi.floris.farm.yaw_angles == yaw_angles
 
 
+def test_calculate_no_wake():
+    """
+    In FLORIS v3.2, running calculate_no_wake twice incorrectly set the yaw angles when the first
+    time has non-zero yaw settings but the second run had all-zero yaw settings. The test below
+    asserts that the yaw angles are correctly set in subsequent calls to calculate_wake.
+    """
+    fi = FlorisInterface(configuration=YAML_INPUT)
+    yaw_angles = 20 * np.ones(
+        (
+            fi.floris.flow_field.n_wind_directions,
+            fi.floris.flow_field.n_wind_speeds,
+            fi.floris.farm.n_turbines
+        )
+    )
+    fi.calculate_no_wake(yaw_angles=yaw_angles)
+    assert fi.floris.farm.yaw_angles == yaw_angles
+
+    yaw_angles = np.zeros(
+        (
+            fi.floris.flow_field.n_wind_directions,
+            fi.floris.flow_field.n_wind_speeds,
+            fi.floris.farm.n_turbines
+        )
+    )
+    fi.calculate_no_wake(yaw_angles=yaw_angles)
+    assert fi.floris.farm.yaw_angles == yaw_angles
+
+    # check finalized values of calculate_no_wake
+    fi_base = FlorisInterface(configuration=YAML_INPUT)
+    yaw_angles = np.zeros(
+        (
+            fi_base.floris.flow_field.n_wind_directions,
+            fi_base.floris.flow_field.n_wind_speeds,
+            fi_base.floris.farm.n_turbines
+        )
+    )
+    fi_base.floris.farm.yaw_angles = yaw_angles
+    fi_base.floris.flow_field.initialize_velocity_field(fi_base.floris.grid)
+    fi_base.floris.farm.initialize(fi_base.floris.grid.sorted_indices)
+    fi_base.floris.state.INITIALIZED
```

Comment:
I don't think we need this here.
Also, this isn't doing anything in `Floris`, so that's a good side effect of including it here! It's accessing the data but not writing it to anything.

Reply:
I removed line 86 above, but the rest is needed to properly setup the `fi_base` object for comparison.